packages/flutter_tools/lib/src/fuchsia/fuchsia_pm.dart

```diff
@@ -180,19 +180,22 @@ class FuchsiaPM {
 ///   server.stop();
 /// }
 class FuchsiaPackageServer {
-  FuchsiaPackageServer(this._repo, this.name, this._host, this._port);
+  FuchsiaPackageServer(this._repo, this.name, this._host, this._port, this._ipv6);
 
   static const String deviceHost = 'fuchsia.com';
   static const String toolHost = 'flutter_tool';
 
   final String _repo;
   final String _host;
   final int _port;
+  final bool _ipv6;
 
   Process _process;
 
   /// The URL that can be used by the device to access this package server.
-  String get url => 'http://$_host:$_port';
+  String get url => _ipv6
```

Comment:
Is this performance critical? My personal preference would be to not require a correct ipv6 determination ahead of time.

```dart
String get url => '${Uri(scheme: 'http', host: _host, port: _port)}';
```

One slight difference in behavior, if `_port` is `80` you won't get the `:80` on this String. That's probably an improvement.

Reply:
Oh, thats a lot nicer - thanks @natebosch

Reply:
Done